web/src/main/java/org/cbioportal/web/SampleController.java

```diff
@@ -129,13 +131,17 @@
         }
     }
 
-    @PreAuthorize("hasPermission(#sampleFilter, 'SampleFilter', 'read')")
+    @PreAuthorize("hasPermission(#involvedCancerStudies, 'Collection<CancerStudyId>', 'read')")
     @RequestMapping(value = "/samples/fetch", method = RequestMethod.POST, consumes = MediaType.APPLICATION_JSON_VALUE,
         produces = MediaType.APPLICATION_JSON_VALUE)
     @ApiOperation("Fetch samples by ID")
     public ResponseEntity<List<Sample>> fetchSamples(
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface
+        @RequestAttribute(required = false, value = "involvedCancerStudies") Collection<String> involvedCancerStudies,
+        @ApiIgnore // prevent reference to this attribute in the swagger-ui interface. this attribute is needed for the @PreAuthorize tag above.
```

Comment:
`interceptedSampleFilter` does not seem to be used in the `@PreAuthorize` tag above.

Reply:
@pieterlukasse That is correct. interceptedSampleFilter is the unmodified / uninterpreted / verbatim argument as passed by the user. This JSON object is parsed / used in the handler function business logic that follows. The '@PreAuthorize' annotation receives an extracted set of cancer studies that contain any of these samples. This extraction is done by InvolvedCancerStudyExtractorInterceptor ... returning model module objects (CancerStudy) which can be checked by the security module without passing the SampleFilter object (defined in the web module). That is the essence of this PR - the web module concepts are hidden from the security module, and instead we are passing model module objects (the model module is a shared dependency to all other modules)